### DISCUSSION TITLE
How to enable features within a module in `appsettings.json`?

### DISCUSSION THREAD

**Discussion by xperiandri** [OrchardCMS/OrchardCore#14416] [Q&A] [answered]
I tried declaring
``` F#
[<Feature("OrchardCore.Mvc.HelloFSharp.Feature")>]
type Startup() =
    inherit StartupBase()

    override _.ConfigureServices(services) = ()

    override _.Configure(builder, routes, serviceProvider) = ()
```

And then enabling it with `appsettings.json`

``` JSON
{
  "OrchardCore": {
    "Default": {
      "State": "Running",
      "RequestUrlHost": null,
      "RequestUrlPrefix": null,
      "Features": [
        "OrchardCore.Mvc.HelloWorld",
        "OrchardCore.Mvc.HelloFSharp",
        "OrchardCore.Mvc.HelloFSharp.Feature"
      ]
    }
  },
}
```

but that does not work. Feature startup is not invoked

@jtkech

**Comment by jtkech** [comment]
Are you referencing `OrchardCore.Mvc.HelloFSharp.Feature` in you app project file?

**Reply by xperiandri** [reply]
![image](https://github.com/OrchardCMS/OrchardCore/assets/2365592/bd93a7ac-3457-456e-988d-98efe956bbb1)
![image](https://github.com/OrchardCMS/OrchardCore/assets/2365592/a5e35618-971a-4adc-ba00-2f6ab73db0c4)

**Reply by jtkech** [reply]
Can you try `[Feature(...)]` in place of `[<Feature(...)>]`

**Reply by xperiandri** [reply]
It is an F# syntax. They are equivalent

**Reply by xperiandri** [reply]
They are not discovered
![image](https://github.com/OrchardCMS/OrchardCore/assets/2365592/fb89d2cd-f510-49a9-b743-3d44b1f66bec)

**Reply by jtkech** [reply]
Are both features belonging to the same module?

If so both the main feature (same id as the module) and the other feature should be defined in the manifest file by using the `[assembly: Feature(...` atrribute, the main feature defined first, as we do in our modules when there is more than one feature (the main one).

**Reply by xperiandri** [reply]
Still 0 features discovered
![image](https://github.com/OrchardCMS/OrchardCore/assets/2365592/a67120cd-e45c-43d7-9512-fc16021cf375)

**Reply by xperiandri** [reply]
I don't understand this code. Looks like it makes an empty list with a weird approach 😄
![image](https://github.com/OrchardCMS/OrchardCore/assets/2365592/6be612d1-a89b-4c5a-bc2a-c8b7f7bac4ce)

**Reply by xperiandri** [reply]
OK, now works on my sample

**Reply by jtkech** [reply]
Cool! What was the issue?

**Reply by xperiandri** [reply, marked as answer]
Necessary pieces:
1. `builder.Services.AddOrchardCore().WithTenants().ApplicationServices.AddExtensionManagerHost()`
2. Feature attribute on feature startup
3. Feature assembly level attribute

**Reply by jtkech** [reply]
Normally `AddExtensionManagerHost()` is called by `AddExtensionServices()` which is called by `AddOrchardCore()`.

**Reply by jtkech** [reply]
If `ConfiguredFeaturesShellDescriptorManager` is really used you first can put a break point in its `GetShellDescriptorAsync()` to see if they are part of the returned `ShellDescriptor`.
